app/assets/javascripts/app/modules/personal-key-formatter.js

```diff
@@ -0,0 +1,45 @@
+import { DelimitedTextFormatter } from 'field-kit';
+
+/**
+ * @const
+ * @private
+ */
+const DIGITS_PATTERN = /^[a-zA-Z0-9]*$/;
+
+/**
+ * @extends DelimitedTextFormatter
+ */
+class PersonalKeyFormatter extends DelimitedTextFormatter {
+  constructor() {
+    super('-', true);
+    this.maximumLength = 16 + 3;
+  }
+
+  /**
+   * @param {number} index
+   * @returns {boolean}
+   */
+
+  /* eslint-disable class-methods-use-this */
+  hasDelimiterAtIndex(index) {
+    return index === 4 || index === 9 || index === 14;
```

Comment:
I was trying to think of a way to avoid the hardcoded integers here, but I see the logic requires counting the characters plus the number of hyphens, which makes the arithmetic tricky. So this seems like the least bad way to do this.